src/solver/where/backward_where.cpp

```diff
@@ -0,0 +1,277 @@
+
+    if(is_all_contiguous)
+    {
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(outputGradNumel, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        kernel.kernel_name = "WhereContiguousBackward";
+
+        kernel.l_wk.push_back(xlocalsize);
+        kernel.l_wk.push_back(ylocalsize);
+        kernel.l_wk.push_back(zlocalsize);
+
+        kernel.g_wk.push_back(xgridsize);
+        kernel.g_wk.push_back(ygridsize);
+        kernel.g_wk.push_back(zgridsize);
+
+        result.construction_params.push_back(kernel);
+
+        result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+                decltype(auto) kernel = handle_.Run(kernels.front());
+                decltype(auto) params = raw_params.CastTo<miopen::where::BwdInvokeParams>();
+
+                size_t output_grad_numel = problem.GetOutputGradDesc().GetElementSize();
+
+                kernel(params.condition,
+                       params.outputGrad,
+                       params.inputGrad,
+                       params.otherGrad,
+                       output_grad_numel);
+            };
+        };
+    }
+    else if(is_condition_broadcasted && is_all_broadcasted_contiguous)
+    {
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(cond_contig_size, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        kernel.kernel_name = "WhereConditionBroadcastedContiguousBackward";
+
+        kernel.l_wk.push_back(xlocalsize);
+        kernel.l_wk.push_back(ylocalsize);
+        kernel.l_wk.push_back(zlocalsize);
+
+        kernel.g_wk.push_back(xgridsize);
+        kernel.g_wk.push_back(ygridsize);
+        kernel.g_wk.push_back(zgridsize);
+
+        result.construction_params.push_back(kernel);
+
+        result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+                decltype(auto) kernel = handle_.Run(kernels.front());
+                decltype(auto) params = raw_params.CastTo<miopen::where::BwdInvokeParams>();
+
+                size_t output_grad_numel      = problem.GetOutputGradDesc().GetElementSize();
+                size_t output_grad_off        = 0;
+                size_t condition_off          = 0;
+                size_t input_grad_off         = 0;
+                size_t other_grad_off         = 0;
+                size_t cond_contig_size       = check_broadcasted_contiguous(cond_tv);
+                size_t input_grad_contig_size = check_broadcasted_contiguous(input_grad_tv);
+                size_t other_grad_contig_size = check_broadcasted_contiguous(other_grad_tv);
```

Comment:
You retrieved cond_tv, input_grad_tv, and other_grad_tv outside of the lambda.
Doing this might prevent tv from being updated in the subsequent where operation.
Looking at check_broadcasted_contiguous, different values can result from the same element size depending on the stride.
If a where operation with the same NetworkConfig (same element size) but different check_broadcasted_contiguous follows, there is an issue where the operation may execute with the previous value without updating the tv.
In this case, instead of calling solver.GetSolution, it directly finds the invoker_factory from the map based on the algo name and NetworkConfig, and executes it. At this time, the tv used during the initial creation of the invoker_factory is used.

Reply:
I have changed network config in [this comment](https://github.com/moreh-dev/MIOpen/pull/32#discussion_r1642550697)